sensor/tests/resource/pod/pod_test.go

```diff
@@ -123,3 +126,70 @@ func (s *PodHierarchySuite) Test_ParentlessPodsAreTreatedAsDeployments() {
 			"Should have received four different pod events (3 from nginx-deployment and 1 from nginx-rouge")
 	})
 }
+
+func (s *PodHierarchySuite) Test_DeleteDeployment() {
+	s.testContext.RunBare("Removing a deployment should send empty violation", func(t *testing.T, testC *resource.TestContext, _ map[string]k8s.Object) {
+		deleteDep, err := testC.ApplyFileNoObject(context.Background(), resource.DefaultNamespace, NginxDeployment)
```

Comment:
Suggestion: If you need the deployment ID, I _think_ you can do that by using `ApplyFile` instead, and pass a `*v1.Deployment` reference. You'd have to test, because AFAIU the object passed will only contain the parsed properties from the yaml. But, the object is also passed to the resources API to be be created in k8s. Maybe the ID will be updated on the `*v1.Deployment` and you don't need to use the callback to figure out what is the ID?

Reply:
Good idea. I've changed it accordingly.